biogeochem/EDPhysiologyMod.F90

```diff
@@ -948,117 +948,164 @@ subroutine phenology( currentSite, bc_in )
+
+    ! Add PFT look to account for different PFT rooting depth profiles.
+    pft_drgt_loop: do ft=1,numpft
+
+       ! Update soil moisture information memory (we always track the last 10 days)
+       do i_wmem = numWaterMem,2,-1 !shift memory to previous day, to make room for current day
+          currentSite%liqvol_memory(i_wmem,ft) = currentSite%liqvol_memory(i_wmem-1,ft)
+          currentSite%smp_memory   (i_wmem,ft) = currentSite%smp_memory   (i_wmem-1,ft)
+       end do
 
-    ! Calculate the mean water content over the last 10 days (m3/m3)
-    mean_10day_liqvol = sum(currentSite%water_memory(1:numWaterMem))/real(numWaterMem,r8)
+       ! Find the rooting depth distribution for PFT
+       call set_root_fraction( currentSite%rootfrac_scr, ft, currentSite%zi_soil, &
+                               bc_in%max_rooting_depth_index_col )
+       nlevroot = max(2,min(ubound(currentSite%zi_soil,1),bc_in%max_rooting_depth_index_col))
+
+       ! The top most layer is typically very thin (~ 2cm) and dries rather quickly. Despite
+       ! being thin, it can have a non-negligible rooting fraction (e.g., using 
+       ! exponential_2p_root_profile with default parameters make the top layer to contain
+       ! about 7% of the total fine root density).  To avoid overestimating dryness, we 
+       ! ignore the top layer when calculating the memory.
+       rootfrac_notop = sum(currentSite%rootfrac_scr(2:nlevroot))
+       if ( rootfrac_notop <= nearzero ) then
+          ! Unlikely, but just in case all roots are in the first layer, we use the second
+          ! layer the second layer (to avoid FPE issues).
+          currentSite%rootfrac_scr(2) = 1.0_r8
+          rootfrac_notop              = 1.0_r8
+       end if
 
-    ! In drought phenology, we often need to force the leaves to stay
-    ! on or off as moisture fluctuates...
+       ! Set the memory to be the weighted average of the soil properties, using the
+       ! root fraction of each layer (except the topmost one) as the weighting factor.
+       currentSite%liqvol_memory(1,ft) = sum( bc_in%h2o_liqvol_sl     (2:nlevroot) * &
+                                              currentSite%rootfrac_scr(2:nlevroot) ) / rootfrac_notop
+       currentSite%smp_memory   (1,ft) = sum( bc_in%smp_sl            (2:nlevroot) * &
+                                              currentSite%rootfrac_scr(2:nlevroot) ) / rootfrac_notop
+
+       ! Calculate the mean soil moisture ( liquid volume (m3/m3) and matric potential (mm))
+       !    over the last 10 days
+       mean_10day_liqvol = sum(currentSite%liqvol_memory(1:numWaterMem,ft))/real(numWaterMem,r8)
+       mean_10day_smp    = sum(currentSite%smp_memory   (1:numWaterMem,ft))/real(numWaterMem,r8)
+
+       ! Compare the moisture with the threshold.
+       if ( ED_val_phen_drought_threshold >= 0. ) then
+          ! Liquid volume in reference layer (m3/m3)
+          smoist_below_threshold = mean_10day_liqvol < ED_val_phen_drought_threshold
```

Comment:
how does Fortran parse the < as different from a - ?

Reply:
I set this to be a logical variable, so it's `smoist_below_threshold = .true.` when the 10-day average is less the threshold.